src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -372,36 +340,56 @@ export const UnderlineNav = forwardRef(
                   style={{display: isWidgetOpen ? 'block' : 'none'}}
                 >
                   {actions.map((action, index) => {
-                    const {children: actionElementChildren, ...actionElementProps} = action.props
+                    const {
+                      children: actionElementChildren,
+                      counter,
+                      'aria-current': ariaCurrent,
+                      onSelect,
+                      ...actionElementProps
+                    } = action.props
+
+                    // // This logic is used to pop the selected item out of the menu and into the list when loading the component
+                    if (Boolean(ariaCurrent) && ariaCurrent !== 'false') {
+                      const event = new MouseEvent('click')
+                      !onlyMenuVisible &&
+                        swapMenuItemWithListItem(
+                          action,
+                          index,
+                          // @ts-ignore - not a big deal because it is internally creating an event but ask help
+                          event as React.MouseEvent<HTMLAnchorElement>,
+                          updateListAndMenu,
+                        )
+                    }
+
                     return (
-                      <Box key={actionElementChildren} as="li">
-                        <ActionList.Item
-                          {...actionElementProps}
-                          as={action.props.as || 'a'}
-                          sx={menuItemStyles}
-                          onSelect={(event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>) => {
-                            // When there are no items in the list, do not run the swap function as we want to keep everything in the menu.
-                            !onlyMenuVisible && swapMenuItemWithListItem(action, index, event, updateListAndMenu)
-                            setSelectEvent(event)
-                            closeOverlay()
-                            focusOnMoreMenuBtn()
-                          }}
-                        >
-                          <Box as="span" sx={{display: 'flex', alignItems: 'center', justifyContent: 'space-between'}}>
-                            {actionElementChildren}
-
-                            {loadingCounters ? (
-                              <LoadingCounter />
-                            ) : (
-                              actionElementProps.counter !== undefined && (
-                                <Box as="span" data-component="counter">
-                                  <CounterLabel>{actionElementProps.counter}</CounterLabel>
-                                </Box>
-                              )
-                            )}
-                          </Box>
-                        </ActionList.Item>
-                      </Box>
+                      <ActionList.LinkItem
```

Comment:
Refactor the component to use `ActionList.LinkItem` instead of `ActionList.Item` because it is a navigation component and we want consumers to use this for this purpose and with items being as anchor links not as buttons or any other elements.